src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -827,17 +827,18 @@ public INamedTypeSymbol GetTypeByMetadataName(string fullyQualifiedMetadataName)
         protected abstract INamedTypeSymbol CommonGetTypeByMetadataName(string metadataName);
 
         /// <summary>
-        /// Returns a new INamedTypeSymbol with the given element types and (optional) element names.
+        /// Returns a new INamedTypeSymbol with the given underlying type and (optional) element names.
         /// </summary>
         /// <remarks>
         /// Since VB doesn't support tuples yet, this call will fail in a VB compilation.
+        /// Also, the underlying type needs to be tuple-compatible.
         /// </remarks>
-        public INamedTypeSymbol CreateTupleTypeSymbol(ImmutableArray<ITypeSymbol> elementTypes, ImmutableArray<string> elementNames = default(ImmutableArray<string>))
+        public INamedTypeSymbol CreateTupleTypeSymbol(INamedTypeSymbol underlyingType, ImmutableArray<string> elementNames = default(ImmutableArray<string>))
```

Comment:
Implementation experiments are not "held hostage" by being kept out of the shipping branch. Github branches are a perfectly reasonable place for such experiments to take place.

I understand that you designed it this way and then implemented it this way. If the LDM approves a spec that corresponds to this change, then of course this implementation approach should be considered for integration. Even if such a language approach is approved, however, I do not believe that the *language effect* inherent in this implementation requires this implementation complexity.